aeron-cluster/src/main/java/io/aeron/cluster/client/AeronCluster.java

```diff
@@ -1378,6 +1514,67 @@ public long messageTimeoutNs()
             return CommonContext.checkDebugTimeout(messageTimeoutNs, TimeUnit.NANOSECONDS);
         }
 
+        /**
+         * The timeout to wait for a new leader after noticing a disconnection from the previous one. Upon timeout the
+         * cluster will be considered lost and the client will close. If set to {@link Aeron#NULL_VALUE}, a reasonable
+         * default will be used.
+         *
+         * @param newLeaderTimeoutNs the new leader timeout in nanoseconds or {@link Aeron#NULL_VALUE}.
+         * @return this for a fluent API.
+         */
+        public Context newLeaderTimeoutNs(final long newLeaderTimeoutNs)
+        {
+            if (!(newLeaderTimeoutNs > 0 || newLeaderTimeoutNs == NULL_VALUE))
+            {
+                throw new IllegalArgumentException(
+                    "newLeaderTimeoutNs must be positive or -1, but was " + newLeaderTimeoutNs);
+            }
+            this.newLeaderTimeoutNs = newLeaderTimeoutNs;
+            return this;
+        }
+
+        /**
+         * The timeout to wait for a new leader after noticing a disconnection from the previous one. Upon timeout the
+         * cluster will be considered lost and the client will close.
+         *
+         * @return the new leader timeout in nanoseconds.
+         */
+        public long newLeaderTimeoutNs()
+        {
+            return newLeaderTimeoutNs;
+        }
+
+        /**
+         * Set the timeout to wait for ingress publication to become connected with a new leader. Upon timeout the
+         * cluster will be considered lost and the client will close. If set to {@link Aeron#NULL_VALUE}, a reasonable
+         * default will be used.
+         *
+         * @param newLeaderConnectionTimeoutNs the new leader connection timeout in nanoseconds
+         *                                     or {@link Aeron#NULL_VALUE}.
+         * @return this for a fluent API.
+         */
+        public Context newLeaderConnectionTimeoutNs(final long newLeaderConnectionTimeoutNs)
```

Comment:
I don't think this configuration option is necessary, `messageTimeout` is sufficient.